### DISCUSSION TITLE
更新了以后之前的定理环境全部失效

### DISCUSSION THREAD

**Discussion by TBBnozomi** [nju-lug/NJUThesis#159] [Q&A] [answered]
我之前用的.19的版本，大量使用了内置的lemma, theorem等定理环境。最近更新到最新版本后此前的形如`\begin{lemma}\end{lemma}`的环境全部失效了，要全部自己手动`\newtheorem`。有除了这个以外的解决办法吗

这是我头一次鲜明的感受到主版本号更新带来的副作用

**Comment by TBBnozomi** [comment, marked as answer]
哦，我知道了，主版本号更新的时候把`\njusetup`改了。现在可以正常用了。建议给手册里面加上一条这个，提醒从旧版本文件改过来的同学记得改配置。

> 年轻人的第一次不兼容的主版本号更新

**Comment by atxy-blip** [comment]
实际上是为了方便用户做修改，默认状态下只*储存*了默认的定理类环境而没有*创建*这些环境。只需要在导言区调用 `\njusetup{theorem/define}` 就可以创建这些环境。

v1.0 希望借换届的时间调整接口以优化某些功能。我对造成的不便深表抱歉。

> 主版本号更新的时候把 `\njusetup` 改了

`\njusetup` 这个命令额外提供了简化的形式，是向下兼容的。但键值列表的选项名称做了调整，可以参考新版的 `njuthesis-sample.tex`.
